src/compactor.rs

```diff
@@ -365,11 +359,115 @@ mod tests {
         // todo: test that the db can read the k/vs (once we implement reading from compacted)
     }
 
+    #[tokio::test]
+    async fn test_should_compact_expired_entries() {
+        // given:
+        let insert_clock = Arc::new(TestClock::new());
+        let compaction_clock = Arc::new(TestClock::new());
+
+        let compactor_opts = CompactorOptions {
+            compaction_scheduler: Arc::new(SizeTieredCompactionSchedulerSupplier::new(
+                SizeTieredCompactionSchedulerOptions {
+                    // compact as soon as we have data
+                    min_compaction_sources: 1,
+                    max_compaction_sources: 10,
+                    include_size_threshold: 4.0,
+                },
+            )),
+            ..compactor_options(compaction_clock.clone())
+        };
+        let options = db_options(Some(compactor_opts), insert_clock.clone());
+        let (_, manifest_store, table_store, db) = build_test_db(options).await;
+
+        // ticker time = 0, expire time = 10
+        insert_clock.ticker.store(0, atomic::Ordering::SeqCst);
+        db.put_with_options(
+            &[1; 16],
+            &[b'a'; 64],
+            &WriteOptions::default(),
+            &PutOptions {
+                ttl: Some(Duration::from_millis(100)),
```

Comment:
should this be `10`?

Reply:
I think the comment was wrong... updated!